pkg/kubelet/deviceplugin/utils.go

```diff
@@ -21,41 +21,16 @@ import (
 
 	"k8s.io/api/core/v1"
 	v1helper "k8s.io/kubernetes/pkg/api/v1/helper"
-	pluginapi "k8s.io/kubernetes/pkg/kubelet/apis/deviceplugin/v1alpha1"
 )
 
-func cloneDevice(d *pluginapi.Device) *pluginapi.Device {
-	return &pluginapi.Device{
-		ID:     d.ID,
-		Health: d.Health,
-	}
-
-}
-
-func copyDevices(devs map[string]*pluginapi.Device) []*pluginapi.Device {
-	var clones []*pluginapi.Device
-
-	for _, d := range devs {
-		clones = append(clones, cloneDevice(d))
-	}
-
-	return clones
-}
-
 // IsResourceNameValid returns an error if the resource is invalid or is not an
 // extended resource name.
 func IsResourceNameValid(resourceName string) error {
 	if resourceName == "" {
```

Comment:
Checking for empty resourceName is redundant. I think IsExtendedResourceName already covers this case.

Reply:
The difference is that the error would stress it's empty if we keep the check.
It's OK to remove the check if we don't care about the content of error

Reply:
I think errInvalidResourceName should cover this error case. If we want to give more meaningful debugging info, we may want to change errInvalidResourceName to print resourceName like "the ResourceName \'%s\' is invalid".